### PR TITLE
Custom lib refactor

### PR DESCRIPTION
**Pull request by ilteoood**
#### Checklist

- [X] your branch will not cause merge conflict with `master`
- [X] run `npm test`
- [X] tests are included
- [X] the documentation is updated or intregrated accordingly with your changes
- [X] the code will follow the lint rules and style of the project
- [X] you are not committing extraneous files or sensible data

This PR doesn't include any new feature, just a refactor of the existing codebase.

### CODE REVIEW

File: lib/serviceBuilder.js [hunk 1]
```diff
@@ -48,16 +48,18 @@ function statusCodeError(statusCode, allowedStatusCodes, message) {
...
   if (isJSON) {
-    options.headers = options.headers || {}
-    options.headers['content-type'] = 'application/json;charset=utf-8'
+    options.headers = { 'content-type': 'application/json;charset=utf-8', ...options.headers }
   }
 
   if (!body) {
-    options.headers = options.headers || {}
-    options.headers['content-length'] = '0'
+    options.headers = { 'content-length': '0', ...options.headers }
```
Comment (silversoul93) [inline lib/serviceBuilder.js:55]: You changed the behavior of the function on this line.
Before your change, the `content-length` header was always set to `0`;
now the `content-length` header is set to `0` if the `options.headers` object does not contains it.

Is this change wanted? Or we can risk some regression.
You can adjust it by changing this line into:
```javascript
options.headers = { ...options.headers, `content-length': '0' }
```

File: lib/serviceBuilder.js [hunk 2]
```diff
@@ -48,16 +48,18 @@ function statusCodeError(statusCode, allowedStatusCodes, message) {
 function setBodyRelatedHeaders(body, options) {
   const isJSON = !Buffer.isBuffer(body) && typeof body !== 'string' && !(body && typeof body.pipe === 'function')
   if (isJSON) {
-    options.headers = options.headers || {}
-    options.headers['content-type'] = 'application/json;charset=utf-8'
+    options.headers = { 'content-type': 'application/json;charset=utf-8', ...options.headers }
```
Comment (silversoul93) [inline lib/serviceBuilder.js:51]: Like the other comment on this file, you changed the behavior of the function here too

File: lib/postDecorator.js [hunk 3]
```diff
@@ -135,17 +121,8 @@ function change() {
...
-        ret.body = this.body
-      }
-      if (this.statusCode) {
-        ret.statusCode = this.statusCode
-      }
-      if (this.headers) {
-        ret.headers = this.headers
-      }
-      return ret
+      const { body, statusCode, headers } = { ...this }
```
Comment (silversoul93) [inline lib/postDecorator.js:124]: Why not just
```
const { body, statusCode, headers } = this
```
Is there a reason for the spread operator of the `this`?

File: lib/util.js [hunk 4]
```diff
@@ -23,10 +23,7 @@ function longerThan(len) {
 }
 
 function getUserId(userId) {
-  if (!userId) {
-    return null
-  }
-  return userId
+  return !userId ? null : userId
```
Comment (davidebianchi) [inline lib/util.js:26]:
```suggestion
  return userId || null
```

Also the following function could be changed like this

File: lib/serviceBuilder.js (hunk 1 above)
Comment (ilteoood) [reply lib/serviceBuilder.js]: You're completely right, this was not wanted. I forgot to reverse the order

File: lib/postDecorator.js (hunk 3 above)
Comment (ilteoood) [reply lib/postDecorator.js]: Nope, fixed

File: lib/util.js (hunk 4 above)
Comment (ilteoood) [reply lib/util.js]: I've made the same fix for getClientType

**Comment by ilteoood** [comment]
ping

**Review by fredmaggiowski — APPROVED** [review]
LGTM

**Pull request merged**